packages/components/src/utils/getReferenceLinkHref.ts

```diff
@@ -1,15 +1,35 @@
+import DOMPurify from "isomorphic-dompurify"
+
 import type { IsomerSitemap } from "~/types"
 import { getSitemapAsArray } from "./getSitemapAsArray"
 
-// Convert the given reference link to the actual permalink
-export const getReferenceLinkHref = (
-  referenceLink: string | undefined,
-  sitemap: IsomerSitemap,
-) => {
-  if (!referenceLink) {
+const ELEMENT_ID = "link-id"
+
+// This function returns a sanitized version of the provided URL string
+const getSanitizedLinkHref = (url?: string) => {
+  if (url === undefined) {
     return undefined
   }
 
+  const dirty = document.createElement("a")
+  dirty.setAttribute("href", url)
+  dirty.setAttribute("id", ELEMENT_ID)
+  const clean = DOMPurify.sanitize(dirty, { RETURN_DOM_FRAGMENT: true })
+
+  const sanitizedUrl = clean.getElementById(ELEMENT_ID)?.getAttribute("href")
```

Comment:
might be able to be simplified (without needing a separate `ELEMENT_ID` const):

```ts
const elem = document.createElement("a")
elem.setAttribute("href", url)
DOMPurify.sanitize(elem, { IN_PLACE: true })
const sanitizedUrl = elem.getAttribute("href")
```

Reply:
Ah missed that option, updated in 11d37516985edebd25a933b0943a26e1b2686eaa.